src/components/app-content-switcher/AppContentSwitcher.tsx

```diff
@@ -1,12 +1,36 @@
+import { TypographyProps } from '@mui/material/Typography'
 import { Typography } from '@mui/material'
 import Tooltip from '@mui/material/Tooltip'
 import Switch from '@mui/material/Switch'
 import Stack from '@mui/material/Stack'
 import { defaultStyles } from '~/components/app-content-switcher/AppContentSwitcher.styles'
 
-const AppContentSwitcher = ({ active, handleChange, switchOptions, typographyVariant, styles }) => {
 
-  const renderBlock = (options, active) => (
+interface SwitcherProps {
+    active: boolean;
+    handleChange: () => void;
+    switchOptions: {
+        left?: {
+            text: string;
+            tooltip?: string;
+        };
+        right?: {
```

Comment:
type declaration for left and right is the same, create a type and use it, also can create type for switchOptions:

type switchOption = {
test: string
tooltip?: string
}
type switcOptions = {
left?:  someType
right: someType
}

But, think about better naming

Reply:
refactored